data_structures/src/chain.rs

```diff
@@ -706,13 +706,14 @@ impl FromStr for Hash {
     type Err = HashParseError;
```

Comment:
I think we should remove `HashParseError` altogether and substitute it with `hex::FromHexError`, that way we don't need to convert error types back and forth.

Reply:
Yes, that's my sentiment too.

Reply:
Makes sense, I think the only problem is that `hex::FromHexError` does not implement `Fail` and may not integrate well with our error system. So it will be easier to just define:

```rust
#[derive(Debug, Fail)]
pub enum HashParseError {
    #[fail(display = "Failed to parse hex: {}", _0)]
    Hex(#[cause] hex::FromHexError),
}
```

And add `.map_err(HashParseError::Hex)` everywhere.